### PR TITLE
Call IdeApp.Exit() instead of letting Process to kill it

### PR DESCRIPTION
**Pull request by manish**
* GetCommandArrayInfoValues (Command) returns the arrayInfo values
    of a command as a List<object>
* Added support to pass the dataItem argument with ExecuteCommand
* Ide.GetSolutionConfigurations () to get list of configurations
    available for a solution created by the template

---

Call IdeApp.Exit() instead of letting Process to kill it

    When IdeApp.Exit () is called, MonoDevelop goes through
    proper code paths to properly shut down itself. Going
    through Process.Kill () is not the correct method since
    it does not hit the correct code paths. There were instances
    where csproxy was not exiting and ended up holding a lot of
    memory thereby hitting 15.96/16GB of memory.

    Thanks to Alan for tracking down the issue

---

Update timeouts

    * Package update timeout is now 120s to account for network slowness
    * Build timeout is at 60s to account for complex solutions

### CODE REVIEW

**Comment by manish** [comment]
adding @slluis and @Therzok

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Components.AutoTest/AutoTestClientSession.cs [hunk 1]
```diff
@@ -114,12 +114,27 @@ public void Stop ()
 			if (service != null)
 				service.DetachClient (this);
 			else
+				try {
 				process.Kill ();
+			} catch (InvalidOperationException invalidExp) {
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Components.AutoTest/AutoTestClientSession.cs]: Indent screwed up here. :P

**Comment by Therzok** [comment]
Otherwise, looks good.

**Comment by slluis** [comment]
I don't think this approach is correct. UI tests should "test" the UI, that is, they should do verifications on what is being shown to the user. A call like GetCommandArrayInfoValues() is using the internal api to get a list of configurations, but there is no guarantee that those configurations are being correctly shown. The test should check the list of configurations that is shown in the main menu (Project -> Active configuration) and on the toolbar (although that may need more work since it is a custom widget).

**Comment by manish** [comment]
Would it be possible to cherry-pick merge f0930a1 and e8c9eb7 commits in this case?

**Comment by manish** [comment]
To get a list of configuration, i'll probably goto Solution Options > Configurations and fetch the list shown there.

**Comment by slluis** [comment]
LGTM

**Pull request merged**